_posts/2025-01-10-this-week-in-rails.markdown

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: "Sorted Columns in Schema Dumper, Deprecations, and lots of fixes!"
+categories: news
+author: vipulnsward
+og_image: assets/images/this-week-in-rails.png
+published: true
+date: 2025-01-10
+---
+
+Hey everyone, Happy Friday!
+
+[Vipul](https://www.saeloun.com/team/vipul) here with the latest updates for This Week in Rails. 
+
+Before we start, a small update- I am happy to share that my team at Saeloun is joining the Rails Foundation as a contributing member.
+Read more about the reasons why we decided to join [here](https://rubyonrails.org/2025/1/10/saeloun-joins-rails-foundation), and why you should too!
+
+[Colorize console prompt on non standard environments](https://github.com/rails/rails/pull/53027)  
+Currently, the Rails console prompt is colored according to the following rules: blue for dev and test, red of prod.
+On non-standard environments (like staging), the prompt is not colored. With this change, it will now be set to Magenta, for these environments.
+
+[Fix running individual app:update commands](https://github.com/rails/rails/pull/54098)  
+This change fixes a regression, where the individual `app:update` commands (like `app:update:configs` or `app:update:bin`) could no longer be run on their own, 
+as they require the app to be loaded and it no longer was.
+
+[Fix backtrace of ArgumentErrors raised during template rendering](https://github.com/rails/rails/pull/54133)  
+Before this fix, any argument error in a template with strict locals would be swallowed and be replaced by an exception with unrelated backtrace, 
+which made debugging templates difficult.
+With this change, any `ArgumentError` unrelated to strict locals is re-raised, preserving the original backtrace for developers.
+
+[Fix local variable access in layouts](https://github.com/rails/rails/pull/54020)  
+Layouts now have access to local variables passed to `render`.
+This fixes a very old regression introduced in Rails 5.1!
+
+[Sort table columns by name when dumping schema](https://github.com/rails/rails/pull/53281)  
+The table columns inside `schema.rb` are now sorted alphabetically.
+Previously they'd be sorted by creation order, which can cause merge conflicts when two branches modify the same table concurrently.
+
+[Update NotificationAssertions's assert_notifcation to match against payload subsets and return matched notification](https://github.com/rails/rails/pull/54126)  
+`ActiveSupport::Testing::NotificationAssertions`'s `assert_notification` now matches against payload subsets by default.
+
+Previously the following assertion would fail due to excess key values in the notification payload. Now with payload subset matching, it will pass-
```

Comment:
```suggestion
Previously the following assertion would fail due to excess key values in the notification payload. Now with payload subset matching, it will pass.
```